bciers/apps/registration/app/components/operations/OperationRegistrationPage.tsx

```diff
@@ -0,0 +1,97 @@
+import { UUID } from "crypto";
+import { validate as isValidUUID } from "uuid";
+import OperationRegistrationForm from "./OperationRegistrationForm";
+import {
+  operationRegistrationSchema,
+  operationRegistrationNewEntrantSchema,
+} from "apps/registration/app/data/jsonSchema/operationRegistration";
+import fetchFacilitiesPageData from "@/administration/app/components/facilities/fetchFacilitiesPageData";
+import {
+  FacilityInitialData,
+  FacilitiesSearchParams,
+} from "@/administration/app/components/facilities/types";
+import { getRegulatedProducts } from "@bciers/actions/api";
+import { RJSFSchema } from "@rjsf/utils";
+
+// 🛠️ Function to create an operation schema with updated enum values
+export const createOperationRegistrationSchema = (
+  schema: RJSFSchema,
+  regulatedProducts: {
+    id: number;
+    name: string;
+  }[],
+) => {
+  const localSchema = JSON.parse(JSON.stringify(schema));
+  const registrationPurposeDependencies =
+    localSchema.properties.registrationPurpose.dependencies;
+  const regulatedProductsSchema =
+    registrationPurposeDependencies.registration_purpose.allOf[0].then
+      .properties.regulated_products;
+
+  // regulated products
+  if (Array.isArray(regulatedProducts)) {
+    regulatedProductsSchema.items.enum = regulatedProducts.map(
+      (product) => product.id,
+    );
+    regulatedProductsSchema.items.enumNames = regulatedProducts.map(
+      (product) => product.name,
+    );
+  }
+
+  return localSchema;
+};
+
+const OperationRegistrationPage = async ({
+  formSection,
+  operation,
+  searchParams,
+}: {
+  formSection: number;
+  operation: UUID | "create";
+  searchParams: FacilitiesSearchParams;
+}) => {
+  // Need to be careful using formSection as it is not a reliable way to determine the page
+  // due to the conditional new entrant page on this form
+  const isFacilityPage = formSection === 3 && operation;
+  const isRegistrationPurposePage = formSection === 1;
+
+  // This will need to be pulled from the database after page 1 is implemented
+  const isNewEntrantOperation = true;
```

Comment:
I meant the dynamic formData that updates as the user fills in the form. Is the idea that we don't show the extra bubble until after the user has submitted?

Reply:
We had discussed in backlog refinement that it wouldn't appear until they submitted and changed pages though I'm not sure that's recorded anywhere. It could be done either way, though I'm not sure how I feel about the bubbles popping in and out based on form changes, it sounds kind of ugly without some animation work.